### PR TITLE
Issue #166 - Adding API call logging

### PR DESCRIPTION
**Pull request by joakime**
Adding info level API call logging using format from python.
Bumps up to warning level for API call failures.

Introduces `RPCFailedStatusException` to allow passing of failed api call status code knowledge.

### CODE REVIEW

File: appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/RPCFailedStatusException.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.google.apphosting.vmruntime;
+
+import com.google.apphosting.api.ApiProxy;
+
+public class RPCFailedStatusException extends ApiProxy.RPCFailedException {
+  private final int statusCode;
+  public RPCFailedStatusException(String packageName, String methodName, int statusCode)
+  {
```
Comment (ludoch) [inline appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/RPCFailedStatusException.java]: Style guide: we usually follow the Google style https://google.github.io/styleguide/javaguide.html

**Comment by joakime** [comment]
will rework.

File: appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmApiProxyDelegate.java [hunk 2]
```diff
@@ -150,15 +151,35 @@ public VmApiProxyDelegate() {
...
     try {
-      return runSyncCall(environment, packageName, methodName, requestData, timeoutMs);
+      byte responseData[] = runSyncCall(environment, packageName, methodName, requestData, timeoutMs);
+      long end = System.currentTimeMillis();
+      logger.log(Level.INFO, String.format(
+              "Service bridge API call to package: %s, call: %s, of size: %s " +
+              "complete. Service bridge status code: %s; response " +
+              "content-length: %s. Took %s ms.",
+              packageName, methodName, requestData.length, 200,
+              responseData.length, (end - start)));
```
Comment (ludoch) [inline appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmApiProxyDelegate.java]: Do you have a link in the console for such a log content? That would help as well.

**Comment by ludoch** [comment]
Thanks!. And a unit test to adapt to the new exception.

File: appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmApiProxyDelegate.java (hunk 2 above)
Comment (joakime) [inline appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmApiProxyDelegate.java]: Not sure how to link to a log ...
It looks like this though

```json
{
    metadata: 
    {
        severity: "INFO"
        projectId: "jetty9-work"
        serviceName: "appengine.googleapis.com"
        zone: "us-central1-c"
        labels:
        {
            appengine.googleapis.com/version_id: "20160217t111117"
            appengine.googleapis.com/thread_id: "qtp1029991479-17"
            compute.googleapis.com/resource_type: "instance"
            compute.googleapis.com/resource_name: "4110d4a7da89"
            compute.googleapis.com/resource_id: "10039582112789601754"
            appengine.googleapis.com/module_id: "default"
        }
        timestamp: "2016-02-17T18:41:01.839Z"
        projectNumber: "170642172088"
    }
    textPayload: "com.google.apphosting.vmruntime.VmApiProxyDelegate makeApiCall: Service bridge API call to package: user, call: CreateLogoutURL, of size: 16 complete. Service bridge status code: 200; response content-length: 258. Took 3 ms."
    insertId: "2016-02-17|10:41:03.618157-08|10.106.13.215|1023972568"
    log: "appengine.googleapis.com/app"
}
```

**Comment by joakime** [comment]
```
[ERROR] Failed to execute goal com.spotify:docker-maven-plugin:0.3.2:build (build) on project openjdk8: Exception caught: Got HTTP status code >= 400: 404 Not Found -> [Help 1]
```

Looks like the openjdk build is failing in docker again, the libssl.deb changed.

```
$ curl --head http://ftp.us.debian.org/debian/pool/main/o/openssl/libssl1.0.2_1.0.2e-1_amd64.deb
HTTP/1.1 404 Not Found
Date: Wed, 17 Feb 2016 21:32:34 GMT
Server: Apache
Content-Type: text/html; charset=iso-8859-1
```

**Comment by joakime** [comment]
Disregard, my branch `bugs/166` is not up to date.

**Comment by ludoch** [comment]
Thanks, LGTM. Will test it on my side as well.

**Comment by gregw** [comment]
Link to issue #166

**Pull request merged**